plugin/metadata/metadataer.go

```diff
@@ -12,10 +12,12 @@ import (
 // Note: this method should work quickly, because it is called for every request
 // from the metadata plugin.
 type Metadataer interface {
-	// Metadata is expected to return map with metadata information which can be
-	// later retrieved from context by any other plugin. It may return empty
-	// map if no metadata needs to be published.
-	Metadata(context.Context, dns.ResponseWriter, *dns.Msg) map[string]interface{}
+	// List of variables which are provided by current metadataer. Must remain constant.
+	MetadataVarNames() []string
+	// Metadata is expected to return a value with metadata information by the key
+	// from first argument. Value can be later retrieved from context by any other plugin.
+	// If value is not available by some reason returned boolean value should be false.
+	Metadata(string, context.Context, dns.ResponseWriter, *dns.Msg) (interface{}, bool)
```

Comment:
There is a linter warning here: "context.Context should be the first parameter of a function (golint)".
Should I move a key from first to the last argument?